contracts/strategies/SnowballProxyStrategyForLPV1.sol

```diff
@@ -0,0 +1,235 @@
+        setAllowances();
+        updateMinTokensToReinvest(_minTokensToReinvest);
+        updateAdminFee(_adminFeeBips);
+        updateDevFee(_devFeeBips);
+        updateReinvestReward(_reinvestRewardBips);
+        updateDepositsEnabled(true);
+        transferOwnership(_timelock);
+
+        emit Reinvest(0, 0);
+    }
+
+    /**
+     * @notice Initialization helper for Pair deposit tokens
+     * @dev Checks that selected Pairs are valid for trading reward tokens
+     * @dev Assigns values to swapPairToken0 and swapPairToken1
+     */
+    function assignSwapPairSafely(address _swapPairWAVAXSnob, address _swapPairToken0, address _swapPairToken1, address _rewardToken) private {
+        require(
+            DexLibrary.checkSwapPairCompatibility(IPair(_swapPairWAVAXSnob), address(WAVAX), address(_rewardToken)),
+            "_swapPairWAVAXSnob is not a WAVAX-Snob pair"
+        );
+        require(
+            _swapPairToken0 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken0), address(WAVAX), IPair(address(depositToken)).token0()),
+            "_swapPairToken0 is not a WAVAX+deposit token0"
+        );
+        require(
+            _swapPairToken1 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken1), address(WAVAX), IPair(address(depositToken)).token1()),
+            "_swapPairToken0 is not a WAVAX+deposit token1"
+        );
+        // converts Snob to WAVAX
+        swapPairWAVAXSnob = IPair(_swapPairWAVAXSnob);
+        // converts WAVAX to pair token0
+        swapPairToken0 = IPair(_swapPairToken0);
+        // converts WAVAX to pair token1
+        swapPairToken1 = IPair(_swapPairToken1);
+    }
+
+    function setSnowballProxy(address _snowballProxy) external onlyOwner {
+        proxy = ISnowballProxy(_snowballProxy);
+    }
+
+    function setAllowances() public override onlyOwner {
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "SnowballStrategyForLPV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(proxy), amount));
+        uint lpAmountDeposited = _stakeDepositTokens(amount);
```

Comment:
lpAmountDeposited can vary because of share math, so here we credit user with slightly less deposit tokens